helix-term/src/commands/lsp.rs

```diff
@@ -83,6 +85,26 @@ fn lsp_location_to_location(
     })
 }
 
+fn deduplicate_locations(locations: Vec<Location>) -> Vec<Location> {
+    let mut map: IndexMap<(Uri, lsp::Range), Location> = IndexMap::new();
+
+    for location in locations {
+        let key = (location.uri.clone(), location.range);
+        match map.get_mut(&key) {
+            Some(existing) => {
+                if location.offset_encoding > existing.offset_encoding {
+                    *existing = location;
+                }
```

Comment:
This handling of offset encoding here is not correct. Offset encoding controls how the `character` offset in a `lsp::Position` (within `lsp::Range`, within `Location`) should be interpreted: either as a byte offset (UTF-8), UTF-16 code unit offset or character offset (UTF-32). We can't know how a `Location` using one offset encoding compares to a `Location` using another offset encoding until we read the file's contents.

Instead let's not attempt to deduplicate locations using different offset encodings, i.e. only deduplicate when the full `Location` is equal

Reply:
Also, considering this, I used the highest `OffsetEncoding` as the default, but we could just as well use the first one that appears, if no difference is actually detectable.

Reply:
Instead of converting the file, couldn't we just check for its encoding?

When dealing with multiple locations to the same line, we would prefer the LSP that is using the same encoding as the file. If there are no LSPs using the file encoding, we simply use the first one.

I will adapt the code to use `HashSet` and `Vec::retain` instead of `IndexSet`.

Reply:
The language server's offset encoding is not related to the file's encoding (confusingly). Offset encoding is chosen by the language server arbitrarily.

Reply:
Is it viable to convert the LSP offset encoding to a default enconding, e. g. UTF-8, to easily handle them?

Then, we could continue to store the LSP's original offset enconding in case another conversion (from UTF-8 to the original encoding) is needed?

Reply:
We can convert between any two offset encodings but we need the file contents to be able to convert